src/main/java/com/aws/greengrass/componentmanager/plugins/docker/DefaultDockerClient.java

```diff
@@ -32,6 +33,17 @@
 public class DefaultDockerClient {
     public static final Logger logger = LogManager.getLogger(DefaultDockerClient.class);
 
+    /**
+     * connect error messages.
+     */
+    private static final String READ_CONNECTION_TIME_OUT = "read: connection timed out";
+    private static final String NET_HTTP_TIMEOUT = "net/http";
+    private static final String TEMPORARY_FAILURE_IN_NAME_RESOLUTION = "Temporary failure in name resolution";
+    private static final String REQUEST_CANCELED = "request canceled";
+    private static final String DOCKER_PULL_TIMEOUT = "timeout";
+    private static final String NO_SUCH_HOST = "no such host";
```

Comment:
Will all these errors be always eventually fixed on retries? Can any of them be permanent errors due to say, the customer's network/proxy settings that can only be fixed by the customer resolving those issues unrelated to Greengrass deployments?

Reply:
> Will all these errors be always eventually fixed on retries? Can any of them be permanent errors due to say, the customer's network/proxy settings that can only be fixed by the customer resolving those issues unrelated to Greengrass deployments?

I think all network issues have nothing matter of Greengrass. So I think these retrying are like waiting the network to be fine and than continue; I also think a finite retry would be better before. Maybe I can change all of these a  finite retry, just tell me the count. Thank you.